ruby/lib/cucumber/ci_environment.rb

```diff
@@ -54,13 +56,19 @@ def detect_git(ci_environment, env)
     def detect_revision(ci_environment, env)
       if env['GITHUB_EVENT_NAME'] == 'pull_request'
         raise StandardError('GITHUB_EVENT_PATH not set') unless env['GITHUB_EVENT_PATH']
-        event = JSON.parse(IO.read(env['GITHUB_EVENT_PATH']))
-        revision = event['pull_request']['head']['sha'] rescue nil
+
+        event = JSON.parse(File.read(env['GITHUB_EVENT_PATH']))
+        revision = begin
+          event['pull_request']['head']['sha']
```

Comment:
Perhaps we can use `event.dig('pull_request, 'head', 'sha')` to remain on on line, without exception handling?

Reply:
Yes I think that would work better here instead of trying to catch a nil

Reply:
Refactored it to dig and seems to work fine